### PR TITLE
Microscope test fix

### LINKED ISSUE #4579: CI:  k8s-1.10.K8sValidatedMicroscope Runs microscope
```
Expected command: kubectl exec -n kube-system microscope -- pgrep -f microscope
To have failed, but it was successful:
Exitcode: 0 
Stdout:
 	 5
	 32
```

[f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip](https://github.com/cilium/cilium/files/2118890/f0c5341e_K8sValidatedMicroscope_Runs_microscope.zip)

### PR DESCRIPTION
**Pull request by nebril**
Evenetually clause is used to test microscope process existence

This PR needs several test runs to make sure microscope flake is fixed

Fixes: #4579 

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4555)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: test/k8sT/microscope.go [hunk 1]
```diff
@@ -56,14 +55,19 @@ var _ = Describe(microscopeTestName, func() {
 		microscopeErr, microscopeCancel := kubectl.MicroscopeStart()
 		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
 
-		time.Sleep(time.Second * 10)
+		Eventually(func() bool {
```
Comment (eloycoto) [inline test/k8sT/microscope.go]: Hi, 

Please don't use Eventually, it does not play nice with BeforeAll and AfterAll, please use `helpers.WithTimeout()` instead.

**Review by eloycoto — DISMISSED** [review]

File: test/k8sT/microscope.go (hunk 1 above)
Comment (nebril) [reply test/k8sT/microscope.go]: Done

**Comment by nebril** [comment]
test-me-please

File: test/k8sT/microscope.go [hunk 2]
```diff
@@ -56,14 +53,28 @@ var _ = Describe(microscopeTestName, func() {
 		microscopeErr, microscopeCancel := kubectl.MicroscopeStart()
 		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
 
-		time.Sleep(time.Second * 10)
+		err := helpers.WithTimeout(func() bool {
+			res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
+			return res.WasSuccessful()
+		}, "failed to start microscope",
```
Comment (ianvernon) [inline test/k8sT/microscope.go]: Error message is somewhat cryptic in relation to what this is checking for. I'd recommend something like "error finding running mircoscope processes" ( ? )

File: test/k8sT/microscope.go [hunk 3]
```diff
@@ -56,14 +53,28 @@ var _ = Describe(microscopeTestName, func() {
...
 		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
 
-		time.Sleep(time.Second * 10)
-
-		res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
-		res.ExpectFail()
+		err = helpers.WithTimeout(func() bool {
+			res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
+			return !res.WasSuccessful()
+		}, "failed to close microscope",
```
Comment (ianvernon) [inline test/k8sT/microscope.go]: Similar to above: error message is somewhat cryptic in relation to what this is checking for. I'd recommend something like "found running microscope processes; no microscope processes should be running" ( ? )

**Review by ianvernon — DISMISSED** [review]

File: test/k8sT/microscope.go (hunk 3 above)
Comment (nebril) [reply test/k8sT/microscope.go]: done

File: test/k8sT/microscope.go (hunk 2 above)
Comment (nebril) [reply test/k8sT/microscope.go]: done

**Comment by nebril** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by aanm** [comment]
Possible flake? https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/4310

**Comment by aanm** [comment]
test-me-please

**Comment by nebril** [comment]
will try one more time, but it seems like flakes flocked to this PR

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
tests passed, let's check again

**Comment by nebril** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 4]
```diff
@@ -445,6 +445,7 @@ func (kub *Kubectl) MicroscopeStart() (error, func() error) {
 
 	cb := func() error {
 		cancel()
+		<-ctx.Done()
```
Comment (eloycoto) [inline test/helpers/kubectl.go:448]: Why this is needed? cancel should be enough no?

**Review by ianvernon — APPROVED** [review]

**Pull request merged**